tools/bin/mbedtls-rewrite-branch-style

```diff
@@ -0,0 +1,586 @@
+
+        Categories:
+        * 'heads' = local branch
+        * 'remotes' = remote name slash remote branch
+        * 'sha' = sha of a commit
+        * 'tags' = tag
+        """
+        if category == 'sha':
+            try:
+                # Don't accidentally pass an option to 'git rev-parse'
+                # (it doesn't support --).
+                if not ref.startswith('-'):
+                    if self.run_git(['rev-parse', ref]).strip() == ref:
+                        return True
+                return False
+            except subprocess.CalledProcessError:
+                # Not a valid ref. It's not our job to raise an exception.
+                return False
+        path = 'refs/{}/{}'.format(category, ref)
+        return self.check_git(['show-ref', '--quiet', '--verify', path])
+
+    def ref_looks_ok_for_template(self, ref: str) -> bool:
+        """Whether ref looks ok to build another branch name on.
+
+        It must be a branch, tag or sha.
+        In particular, reject HEAD.
+        """
+        if self.is_git_ref('heads', ref):
+            return True
+        if self.is_git_ref('remotes', ref):
+            return True
+        if self.is_git_ref('tags', ref):
+            return True
+        if self.is_git_ref('sha', ref):
+            return True
+        return False
+
+    def fill_branch_name_template(self, template: str, ref: str,
+                                  kind: str) -> str:
+        """Build a branch name based on a template.
+
+        Replace '{}' by ref if present (otherwise return template unchanged).
+        ref must be a branch, tag or sha.
+        kind is a short, human-readable description of the template's purpose.
+        """
+        if '{}' not in template:
+            return template
+        if not self.ref_looks_ok_for_template(ref):
+            self.log_error('Cannot create branch name for {} from {}',
+                           kind, ref)
+            raise ValueError('Bad ref {} for {} branch name template {}'
+                             .format(ref, kind, template))
+        return template.replace('{}', ref)
+
+    def working_directory_is_clean(self) -> bool:
+        """Check whether the Git working directory is clean (no uncommited changes)."""
+        return (self.check_git(['diff', '--quiet']) and
+                self.check_git(['diff', '--quiet', '--cached']))
+
+    def list_commit_range(self, start: str, last: str) -> List[str]:
+        """List the commits from start to last, excluding last."""
+        commits = self.get_git_lines(['rev-list', start + '..' + last])
+        return list(reversed(commits))
+
+    def system_is_ok(self) -> bool:
+        """Check that the required tools are present and the working directory is suitable."""
+        ok = True
+        # 'git worktree remove' was added in Git 2.17.0. It's the most
+        # recent Git feature that we need.
+        output = self.run_git(['worktree', '--help'])
```

Comment:
This appears to not work as intended with windows displaying `--help` commands throught a browser. Would it make sense to specifically check for the verison?

~~~~~~
output = self.run_git(['version'])
try:
    map(int, re.findall(r'git version (\d+.\d+)', v)[0].split("."))
    if major_v ==2 and minor_v < 17:
       ok = False
except Exception:
   ok =  False
~~~~~~

Reply:
Fixed in https://github.com/Mbed-TLS/mbedtls-docs/pull/73